### ISSUE TITLE
ERROR [ExceptionsHandler] [big.js] Division by zero

### ISSUE THREAD

**Issue by TrikkStar** [ghostfolio/ghostfolio#1209]
I'm currently self-hosting this on unRaid and can't seem to get it to work. Spun up a new instance following the instructions, added a few test activities to the default account, and went to the Holdings page where I get the following error in the log:
```
[Nest] 409  - 08/31/2022, 10:10:08 AM   ERROR [ExceptionsHandler] [big.js] Division by zero
Error: [big.js] Division by zero
    at Big.P.div (/ghostfolio/apps/api/node_modules/big.js/big.js:356:13)
    at PortfolioService.<anonymous> (/ghostfolio/apps/api/main.js:1:364784)
    at Generator.next (<anonymous>)
    at fulfilled (/ghostfolio/apps/api/node_modules/tslib/tslib.js:115:62)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by dtslvr** [comment]
Hi @TrikkStar

Could you export your activities and upload the file for further analysis? Thank you.

**Comment by TrikkStar** [comment]
Sure thing, I also get the same error when trying to use the Live Demo (which seems mostly broken in my instance)
```
{
  "meta": {
    "date": "2022-09-01T12:38:12.005Z",
    "version": "v1.185.0"
  },
  "activities": [
    {
      "accountId": "60978dc6-c1c9-416d-9175-d50a8f839e5d",
      "comment": null,
      "fee": 0,
      "quantity": 1,
      "type": "BUY",
      "unitPrice": 32.14,
      "currency": "USD",
      "dataSource": "YAHOO",
      "date": "2017-12-20T05:00:00.000Z",
      "symbol": "SPDW"
    },
    {
      "accountId": "60978dc6-c1c9-416d-9175-d50a8f839e5d",
      "comment": null,
      "fee": 0,
      "quantity": 3,
      "type": "BUY",
      "unitPrice": 64.01,
      "currency": "USD",
      "dataSource": "YAHOO",
      "date": "2017-12-20T05:00:00.000Z",
      "symbol": "ILTB"
    },
    {
      "accountId": "60978dc6-c1c9-416d-9175-d50a8f839e5d",
      "comment": null,
      "fee": 0,
      "quantity": 6,
      "type": "BUY",
      "unitPrice": 32,
      "currency": "USD",
      "dataSource": "YAHOO",
      "date": "2017-12-19T05:00:00.000Z",
      "symbol": "SPDW"
    },
    {
      "accountId": "60978dc6-c1c9-416d-9175-d50a8f839e5d",
      "comment": null,
      "fee": 0,
      "quantity": 1,
      "type": "BUY",
      "unitPrice": 37.3,
      "currency": "USD",
      "dataSource": "YAHOO",
      "date": "2017-12-19T05:00:00.000Z",
      "symbol": "SPEM"
    },
    {
      "accountId": "60978dc6-c1c9-416d-9175-d50a8f839e5d",
      "comment": null,
      "fee": 6.95,
      "quantity": 4,
      "type": "BUY",
      "unitPrice": 44.98,
      "currency": "USD",
      "dataSource": "YAHOO",
      "date": "2017-12-19T05:00:00.000Z",
      "symbol": "VEA"
    }
  ]
}
```

**Comment by tezgno** [comment]
Seeing the same error on a fresh install as well.

**Comment by dtslvr** [comment]
Thank you @TrikkStar.

It looks like this happens when the data provider does not send data. I was able to reproduce this error while I was disconnected from the Internet.

I have created a fix which improves the error handling (#1215).
